lib/files.c

```diff
@@ -37,47 +37,59 @@ static bool get_file_size(FILE *fp, long *file_size, const char *path) {
     return true;
 }
 
-bool files_load_bytes_from_file(const char *path, UINT8 *buf, UINT16 *size) {
-
-    if (!buf || !size || !path) {
-        return false;
-    }
-
-    FILE *f = fopen(path, "rb");
-    if (!f) {
-        LOG_ERR("Could not open file \"%s\" error %s", path, strerror(errno));
-        return false;
-    }
-
+static bool read_bytes_from_file(FILE *f, UINT8 *buf, UINT16 *size,
+                                 const char *path) {
     long file_size;
     bool result = get_file_size(f, &file_size, path);
     if (!result) {
         /* get_file_size() logs errors */
-        goto err;
+        return false;
     }
 
     /* max is bounded on UINT16 */
     if (file_size > *size) {
         LOG_ERR(
                 "File \"%s\" size is larger than buffer, got %ld expected less than %u",
                 path, file_size, *size);
-        goto err;
+        return false;
     }
 
     result = files_read_bytes(f, buf, file_size);
     if (!result) {
         LOG_ERR("Could not read data from file \"%s\"", path);
-        goto err;
+        return false;
     }
 
     *size = file_size;
-    /* result set on files_read_bytes */
 
-err:
+    return true;
+}
+
+bool files_load_bytes_from_file(const char *path, UINT8 *buf, UINT16 *size) {
+    if (!buf || !size || !path) {
+        return false;
+    }
+
+    FILE *f = fopen(path, "rb");
+    if (!f) {
+        LOG_ERR("Could not open file \"%s\" error %s", path, strerror(errno));
+        return false;
+    }
+
+    bool result = read_bytes_from_file(f, buf, size, path);
+
     fclose(f);
     return result;
 }
 
+bool files_load_bytes_from_stdin(UINT8 *buf, UINT16 *size) {
+    if (!buf || !size) {
```

Comment:
I'm hoping at one point in time, that a lot of this can go through another round of refactoring as the tools mature. I just pulled your changes locally to look through. LGTM.